File: sprint1/By languages/C++/lesson7code4.cpp
#include <iostream>
#include <vector>
using namespace std;

vector<bool> eratosthenes(int n) {
    vector<bool> numbers(n + 1, true);
    numbers[0] = numbers[1] = false;
    for (int num = 2; num < n; num++) {
        if (numbers[num]) {
            for (int j = 2 * num; j <= n; j += num) {
                numbers[j] = false;
            }
        }
    }
    return numbers;
}

File: sprint1/By languages/swift/lesson7code4.swift
func eratosthenes(_ n: Int) -> [Bool] {
    var numbers = Array(repeating: true, count: n + 1)
    numbers[0] = false
    numbers[1] = false
    for num in 2..<n {
        if numbers[num] {
            var j = 2 * num
            while j <= n {
                numbers[j] = false
                j += num
            }
        }
    }
    return numbers
}